workers/unity/Packages/io.improbable.gdk.buildsystem/Util/BuildSupportChecker.cs

```diff
@@ -1,11 +1,31 @@
+using System.Collections.Generic;
 using System.IO;
+using System.Linq;
 using Improbable.Gdk.BuildSystem.Configuration;
 using UnityEditor;
 using UnityEngine;
 
 public static class BuildSupportChecker
 {
+    private static HashSet<BuildTarget> supportedBuildTargets;
+
+    private static void CheckBuildSupportTargets()
+    {
+        supportedBuildTargets = new HashSet<BuildTarget>(WorkerBuildData.AllBuildTargets.Where(CheckForBuildSupport));
+    }
+
+
     public static bool CanBuildTarget(BuildTarget target)
+    {
+        if (supportedBuildTargets == null)
+        {
+            CheckBuildSupportTargets();
```

Comment:
This could be moved into a static constructor so we know that the `HashSet` is always built by the time we enter this function

Reply:
Explicitly didn't do this. Since I don't want this to run every time a domain reload happens.
Just to save a bit of time.

Reply:
Does the HashSet serialize during the domain reload?

Reply:
No

Reply:
Then you have to run this every domain reload anyway? Static constructors are lazily called I think (although unsure how that interacts with Unity)